controllers/rollingupgrade_controller.go

```diff
@@ -1000,8 +1001,7 @@ func (r *RollingUpgradeReconciler) UpdateInstance(ctx *context.Context,
 	// If the running node has the same launchconfig as the asg,
 	// there is no need to refresh it.
 	targetInstanceID := aws.StringValue(i.InstanceId)
-	if !requiresRefresh(i, launchDefinition) {
-		r.info(ruObj, "Ignoring since it has the correct launch-config", "instanceID", targetInstanceID)
+	if !r.requiresRefresh(ruObj, i, launchDefinition) {
```

Comment:
Won't this mean if the controller gets restarted and reloads the specs it will refresh again?
we should set it to false once we are done, or use an annotation and just delete it.

Reply:
Great point ..

Reply:
One option is to annotate all the existing nodes if the `forceRefresh` flag is set and then look for whether the annotation exists when the node is being processed. This has a couple of drawbacks. (i) If there are large number of nodes in the ASG, all these will require an update thereby adding that much more work to do on the APIServer and (ii) the implementation will be fragmented between processing the `forceRefresh` flag initially and annotating all the nodes and then checking for each one of the nodes.

How about this:

- Every object in Kubernetes has a `creationTimestamp`. The node object has a creationTimestamp and the CR itself also has a creationTimestamp.
- Within `refreshRequired`, if the `forceRefresh` flag is set, there can be an additional check that the `creationTimestamp` of the node should be less than the `creationTimeStamp` of the CR. If the node was created after the CR was created, it won't be part of the refresh.

WDYT?

Reply:
👍 I think that should work.